apps/src/lib/kits/maker/Button.js

```diff
@@ -19,4 +19,25 @@ export default function Button(opts) {
     get: () => this.value === 1
   });
 }
-Button.inherits(five.Button);
+PlaygroundButton.inherits(five.Button);
+
+export function MicroBitButton(board) {
+  // There are six button events, ['', 'down', 'up', 'click', 'long-click', 'hold']
+  this.buttonEvents = new Array(6).fill(0);
+  board.mb.addFirmataEventListener((sourceID, eventID) => {
+    if (board.pin === sourceID) {
+      this.buttonEvents[eventID]++;
```

Comment:
What does this do? When is it decremented?

Reply:
Each type of button event is initialized at zero, so this counts the number of times an event has occurred. Whether this needs to be constantly counted upwards or whether we could just increment from zero to one (i.e. not yet pressed to has been pressed), is something I don't have a strong preference on.